### PR TITLE
Improve the Readme

### PR DESCRIPTION
**Pull request by erbsenzaehler**
I tried to improve the Readme because many people in my local chess club do not understand some of the uci options properly. Starting point of this was Cfish's Readme, some internet resources and the code itself.

### CODE REVIEW

**Comment by mcostalba** [comment]
Thanks! A great job!

File: Readme.md [hunk 1]
```diff
@@ -1,66 +1,100 @@
...
+    Clear the hash table.
+
+  * #### Ponder
+    Let Stockfish ponder its next move while the opponent is thinking.
+
+  * #### MultiPV
+    Output the N best lines (principle variantions, PVs) when searching. Leave at 1 for best performance.
+
+  * #### Skill Level
+    Lower the Skill Level in order make Stockfish play weaker.
```
Comment (Alayan-stk-2) [inline Readme.md]: Missing word

File: Readme.md [hunk 2]
```diff
@@ -96,23 +130,30 @@ compile (for instance with Microsoft MSVC) you need to manually
...
+
+Improving Stockfish requires a massive amount of testing. You can simply donate your hardware by installing the [Fishtest Worker](https://github.com/glinscott/fishtest/wiki/Running-the-worker) and view the current tests on [Fishtest](http://tests.stockfishchess.org/tests).
+
+### Improving the code
+
+If you want to help to improve the code, there are several valuable ressources to read:
+
+* [In this wiki,](https://www.chessprogramming.org) many techniques used in Stockfish are explained with a lot of background information.
+
+* [The section on Stockfish](https://www.chessprogramming.org/Stockfish) describes many features and techniques used by Stockfish. However, it is generic rather than being focused on Stockfish's precise implementation. Nethertheless, a helpful resource.
```
Comment (Alayan-stk-2) [inline Readme.md]: Typo, should be "Nevertheless"

File: Readme.md [hunk 3]
```diff
@@ -1,66 +1,100 @@
...
+  * Copying.txt, a text file containing the GNU General Public License version 3.
 
   * src, a subdirectory containing the full source code, including a Makefile
     that can be used to compile Stockfish on Unix-like systems.
 
 
-### Syzygybases
+## UCI parameters
+
+Currently Stockfish exposes the following UCI options:
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
Currently, Stockfish has the following UCI options:
```

File: Readme.md [hunk 4]
```diff
@@ -1,66 +1,100 @@
...
+
+  * #### Debug Log File
+    Write all communication from the engine and to the engine into a text file.
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
    Write all communication to and from the engine into a text file.
```

File: Readme.md [hunk 5]
```diff
@@ -1,66 +1,100 @@
...
+
+  * #### Contempt
+    A positive value for contempt favors middle game positions and avoids draws.
+
+  * #### Analysis Contempt
+    By default, contempt is set to prefer the side to move. Set this option to White or Black to analyse with contempt for that side or set this option to Off to disable contempt.
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
    By default, contempt is set to prefer the side to move. Set this option to "White" or "Black" to analyse with contempt for that side, or "Off" to disable contempt.
```

File: Readme.md [hunk 6]
```diff
@@ -1,66 +1,100 @@
...
+
+  * #### Threads
+    The number of CPU threads used for searching a position. Should equal the number of CPU cores for best performance.
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
    The number of CPU threads used for searching a position. For best performance, set this equal to the number of CPU cores available.
```

File: Readme.md [hunk 7]
```diff
@@ -1,66 +1,100 @@
...
-"SyzygyProbeDepth", "Syzygy50MoveRule" and "SyzygyProbeLimit".
+  * #### Minimum Thinking Time
+    Search for at least x ms per move. 
 
-The option "SyzygyPath" should be set to the directory or directories that
-contain the .rtbw and .rtbz files. Multiple directories should be
-separated by ";" on Windows and by ":" on Unix-based operating systems.
-**Do not use spaces around the ";" or ":".**
+  * #### Slow Mover
+    Lower value will make Stockfish take less times in games, higher values will make it think longer.
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
    Lower values will make Stockfish take less time in games, higher values will make it think longer.
```

File: Readme.md [hunk 8]
```diff
@@ -1,66 +1,100 @@
...
+  * #### SyzygyPath
+    Path to the folders/directories storing the Syzygy tablebase files. Multiple directories are to be separated by ";" on Windows and by ":" on Unix-based operating systems. Do not use spaces around the ";" or ":".
+    
+    Example: `C:\tablebases\wdl345;C:\tablebases\wdl6;D:\tablebases\dtz345;D:\tablebases\dtz6`
+    
+    It is recommended to store .rtbw files on an SSD. There is no loss in storing the .rtbz files on a regular HD.
 
-The "SyzygyProbeLimit" option should normally be left at its default value.
+  * #### SyzygyProbeDepth
+    Minimum remaining search depth for which a position is probed. Set this option to a higher value to probe less agressively if you experience too much
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
    Minimum remaining search depth for which a position is probed. Set this option to a higher value to probe less aggressively if you experience too much slowdown (in terms of nps) due to TB probing.
```

File: Readme.md [hunk 9]
```diff
@@ -1,66 +1,100 @@
...
+    slowdown (in terms of nps) due to TB probing.
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
```

File: Readme.md [hunk 10]
```diff
@@ -96,23 +130,30 @@ compile (for instance with Microsoft MSVC) you need to manually
...
 
-### Resource For Understanding the Code Base
 
-* [Chess Programming Wiki](https://www.chessprogramming.org/Main_Page) 
-has good overall chess engines explanations 
-(techniques used here are well explained like hash maps etc), it was 
-also recommended by the [support team at stockfish.](http://support.stockfishchess.org/discussions/questions/1132-how-to-understand-stockfish-sources)
+## Resource For Understanding the Code Base and participating in the project
+
+Stockfish's improvement of the last couple of years have been a great community effort. You can help getting stronger in different ways.
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
Stockfish's improvement of the last couple of years has been a great community effort. There are a few ways to help contribute to its growth.
```

File: Readme.md [hunk 11]
```diff
@@ -96,23 +130,30 @@ compile (for instance with Microsoft MSVC) you need to manually
...
+
+### Donating hardware
+
+Improving Stockfish requires a massive amount of testing. You can simply donate your hardware by installing the [Fishtest Worker](https://github.com/glinscott/fishtest/wiki/Running-the-worker) and view the current tests on [Fishtest](http://tests.stockfishchess.org/tests).
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
Improving Stockfish requires a massive amount of testing. You can donate your hardware resources by installing the [Fishtest Worker](https://github.com/glinscott/fishtest/wiki/Running-the-worker) and view the current tests on [Fishtest](http://tests.stockfishchess.org/tests).
```

File: Readme.md [hunk 12]
```diff
@@ -96,23 +130,30 @@ compile (for instance with Microsoft MSVC) you need to manually
...
+
+### Improving the code
+
+If you want to help to improve the code, there are several valuable ressources to read:
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
If you want to help improve the code, there are several valuable resources:
```

File: Readme.md [hunk 13]
```diff
@@ -96,23 +130,30 @@ compile (for instance with Microsoft MSVC) you need to manually
...
+
+* [In this wiki,](https://www.chessprogramming.org) many techniques used in Stockfish are explained with a lot of background information.
+
+* [The section on Stockfish](https://www.chessprogramming.org/Stockfish) describes many features and techniques used by Stockfish. However, it is generic rather than being focused on Stockfish's precise implementation. Nevertheless, a helpful resource.
 
-* [Here](https://www.chessprogramming.org/Stockfish) you can find a set 
-of features and techniques used by Stockfish and each of them is explained 
-at the wiki, however, it's a generic way rather than focusing on Stockfish's 
-own implementation, but it will still help you.
+* The latest source can always be found on  [GitHub](https://github.com/official-stockfish/Stockfish). Discussions about Stockfish take place in the [FishCooking](https://groups.google.com/forum/#!forum/fishcooking) group and engine testing is done on [Fishtest](http://tests.stockfishchess.org/tests). If you want to help improve Stockfish, please read this [guideline](https://github.com/glinscott/fishtest/wiki/Creating-my-first-test) first, where the fundamentals about Stockfish development are explained.
```
Comment (solarlight2) [inline Readme.md]:
```suggestion
* The latest source can always be found on  [GitHub](https://github.com/official-stockfish/Stockfish). Discussions about Stockfish take place in the [FishCooking](https://groups.google.com/forum/#!forum/fishcooking) group and engine testing is done on [Fishtest](http://tests.stockfishchess.org/tests). If you want to help improve Stockfish, please read this [guideline](https://github.com/glinscott/fishtest/wiki/Creating-my-first-test) first, where the basics of Stockfish development are explained.
```

Comment (erbsenzaehler) [reply Readme.md]: thanks for your suggestions. i implemented them

File: Readme.md [hunk 14]
```diff
@@ -1,66 +1,99 @@
...
+    The size of the hash table in MB.
+
+  * #### Clear Hash
+    Clear the hash table.
+
+  * #### Ponder
+    Let Stockfish ponder its next move while the opponent is thinking.
+
+  * #### MultiPV
+    Output the N best lines (principle variantions, PVs) when searching. Leave at 1 for best performance.
```
Comment (apetresc) [inline Readme.md]: `variations` instead of `variantions`

File: Readme.md [hunk 15]
```diff
@@ -96,23 +129,30 @@ compile (for instance with Microsoft MSVC) you need to manually
 set/unset some switches in the compiler command line; see file *types.h*
 for a quick reference.
 
-### Resource For Understanding the Code Base
 
-* [Chess Programming Wiki](https://www.chessprogramming.org/Main_Page) 
-has good overall chess engines explanations 
-(techniques used here are well explained like hash maps etc), it was 
-also recommended by the [support team at stockfish.](http://support.stockfishchess.org/discussions/questions/1132-how-to-understand-stockfish-sources)
+## Resource For Understanding the Code Base and participating in the project
```
Comment (apetresc) [inline Readme.md]: Really weird title casing here. Headers for the other sections all just use normal sentence-casing, so should be the same here.

File: Readme.md [hunk 16]
```diff
@@ -96,23 +129,30 @@ compile (for instance with Microsoft MSVC) you need to manually
...
+
+Stockfish's improvement of the last couple of years has been a great community effort. There are a few ways to help contribute to its growth.
```
Comment (apetresc) [inline Readme.md]: "over the last couple of years" is more idiomatic.

File: Readme.md (hunk 15 above)
Comment (erbsenzaehler) [reply Readme.md]: Thanks for reading all this stuff. Not sure what you would like to read here, but I made a small modification anyway.

File: Readme.md [hunk 17]
```diff
@@ -49,7 +49,7 @@ Currently, Stockfish has the following UCI options:
     Let Stockfish ponder its next move while the opponent is thinking.
 
   * #### MultiPV
-    Output the N best lines (principle variantions, PVs) when searching. Leave at 1 for best performance.
+    Output the N best lines (principle variations, PVs) when searching. Leave at 1 for best performance.
```
Comment (Alayan-stk-2) [inline Readme.md]: Should be principal variations, not principle variations

**Comment by snicolet** [comment]
@erbsenzaehler 

Would it be possible to add return carriages at about 75-80 characters, like it was on the old readme? 

See the diff for the current state of the pull request, where the left column has much better readability: https://github.com/official-stockfish/Stockfish/pull/1898/files

**Comment by snicolet** [comment]
Merged via 800031c94c655e64e436846911e599de6eb35a72, thanks to all :-)

**Pull request closed without merging**